beacon-chain/das/blocking.go

```diff
@@ -0,0 +1,90 @@
+package das
+
+import (
+	"context"
+	"sync"
+
+	errors "github.com/pkg/errors"
+	fieldparams "github.com/prysmaticlabs/prysm/v4/config/fieldparams"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/blocks"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/primitives"
+	ethpb "github.com/prysmaticlabs/prysm/v4/proto/prysm/v1alpha1"
+)
+
+// BlockingAvailabilityStore wraps NewCachingDBVerifiedStore and blocks until IsDataAvailable is ready.
+// If the context given to IsDataAvailable is cancelled, the result of IsDataAvailable will be ctx.Error().
+type BlockingAvailabilityStore struct {
+	notif *idxNotifiers
+	s     AvailabilityStore
+}
+
+func (bs *BlockingAvailabilityStore) PersistOnceCommitted(ctx context.Context, current primitives.Slot, sc ...*ethpb.BlobSidecar) []*ethpb.BlobSidecar {
+	if len(sc) < 1 {
+		return nil
+	}
+	seen := bs.notif.ensure(keyFromSidecar(sc[0]))
```

Comment:
Should we be defensive and check all the `sc` are similar? Also there's no more than 6